src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -352,10 +350,13 @@ private HttpResponse sendAndDefer(HttpRequest request, HttpRequestOptions option
       Response response = asyncHandler.getResponse();
       HttpResponse httpResponse = httpResponseCreator.create(response, inPipe);
       if (shouldFollowRedirect(httpResponse, options, enableMuleRedirect)) {
-        if (currentRedirects > MAX_REDIRECTS) {
+        if (currentRedirects >= MAX_REDIRECTS) {
```

Comment:
Doesn't this break backwards compatibility?

Reply:
No, I have tested and the correct is to put >= to work as Grizzly does.